src/core/annotation.js

```diff
@@ -3947,18 +3985,27 @@ class FreeTextAnnotation extends MarkupAnnotation {
   }
 
   static createNewDict(annotation, xref, { apRef, ap }) {
-    const { color, fontSize, oldAnnotation, rect, rotation, user, value } =
-      annotation;
+    const {
+      color,
+      date,
+      fontSize,
+      oldAnnotation,
+      rect,
+      rotation,
+      user,
+      value,
+    } = annotation;
+    const pdfDate = `D:${getModificationDate(date)}`;
     const freetext = oldAnnotation || new Dict(xref);
     freetext.setIfNotExists("Type", Name.get("Annot"));
     freetext.setIfNotExists("Subtype", Name.get("FreeText"));
     if (oldAnnotation) {
-      freetext.set("M", `D:${getModificationDate()}`);
+      freetext.set("M", pdfDate);
```

Comment:
Let's use the same structure as for the annotation types changed below for consistency:

```
freetext.set(
  oldAnnotation ? "M" : "CreationDate",
  `D:${getModificationDate(date)}`
);
```

This keeps the `getModificationDate` call in a single place so we also don't need the separate `pdfDate` variable.